remove redundant remote_user for local setting

bcoca opened the pull request:
local action plugin already does and this also should fix
  fork/thread issue by removing use of pwd library

  fixes #59642

(cherry picked from commit 488b9d6c35b22205522c964182480e398e2cf4bc)
##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
play_context

bcoca committed:
remove redundant remote_user for local setting

  local action plugin already does and this also should fix
  fork/thread issue by removing use of pwd library

  fixes #59642

(cherry picked from commit 488b9d6c35b22205522c964182480e398e2cf4bc)

felixfontein commented:
This commit in `devel` breaks `copy` with `delegate_to: localhost` for me (https://github.com/ansible/ansible/pull/72412#issuecomment-723641676).

bcoca commented:
fix in #72543, will add comimt to this PR

bcoca committed:
ensure local exposes correct user (#72543)

* ensure local exposes correct user

  avoid corner case in which delegation relied on
  playcontext fallback which was removed

  fixes #72541

(cherry picked from commit aa4d53ccdfe9fb8f5a97e058703286adfbc91d08)

bcoca commented:
added fix to this PR

felixfontein reviewed (approved):
Works for me.

relrod commented:
/azp run
